unixfs/src/walk.rs

```diff
@@ -356,93 +446,21 @@ impl Walker {
 
 /// Represents what the `Walker` is currently looking at. Converted to `Entry` for public API.
 #[derive(Debug)]
-struct InnerEntry {
+pub(crate) struct InnerEntry {
+    cid: Cid,
     kind: InnerKind,
     path: PathBuf,
     metadata: Metadata,
     depth: usize,
 }
 
-impl From<InnerEntry> for Metadata {
-    fn from(e: InnerEntry) -> Self {
-        e.metadata
-    }
-}
-
-// FIXME: could simplify roots to optinal cid variants?
-enum InnerKind {
-    /// This is necessarily at the root of the walk
-    RootDirectory(Cid),
-    /// This is necessarily at the root of the walk
-    BucketAtRoot(Cid),
-    /// This is the metadata containing bucket, for which we have a name
-    RootBucket(Cid),
-    /// This is a sibling to a previous named metadata containing bucket
-    Bucket(Cid),
-    /// Directory on any level except root
-    Directory(Cid),
-    /// File optionally on the root level
-    File(Cid, Option<FileVisit>, u64),
-    /// Symlink optionally on the root level
-    Symlink(Cid),
-}
-
-impl fmt::Debug for InnerKind {
```

Comment:
Why was this not returned for InnerEntry?

Reply:
`Cid` was moved to `InnerEntry` to avoid yet another matching when transposing to `ContinuedWalk` and since every variant has `Cid`, it doesn't seemed to be a problem.

In regards to debug representations, this lost formatting can be re-implemented to `InnerEntry` in a following PR.